Source/Fuse.Controls.Panels/Layouts/WrapLayout.uno

```diff
@@ -121,21 +121,21 @@ namespace Fuse.Layouts
 			if (_hasItemHeight)
 				clp.SetY(ItemHeight);
 
-			var placements = new float4[elements.Count];
+			var placements = new float4[container.Children.Count];
 			//minorMaxSize in each major row, assinged per element
-			var minorSizes = new float[elements.Count];
+			var minorSizes = new float[container.Children.Count];
 			// save the row each element is on
-			var elementOnRow = new int[elements.Count];
+			var elementOnRow = new int[container.Children.Count];
 			// save the available space for each row
-			var majorRest = new float[elements.Count];
+			var majorRest = new float[container.Children.Count];
 			//where this row starts
 			int majorStart = 0;
 			// current row
 			int currentRow = 0;
 			
-			for (int i = 0; i < elements.Count;++i)
+			int i = 0;
+			for (var e = container.FirstChild<Visual>(); e != null; e = e.NextSibling<Visual>(), i++)
```

Comment:
The variable `i` has a different value than previously. You're only iterating `Visual` now instead of `Node`, thus `i++` is not incremented as often. You probably need to change this to `FirstChild<Node>` and `NextSibling<Node>` just to be safe.